tests/test_core.py

```diff
@@ -1119,6 +1136,21 @@ def test_merge_topk_PI_with_overlap():
         PB[:, :] = np.take_along_axis(PB, IDX, axis=1)
         IB[:, :] = np.take_along_axis(IB, IDX, axis=1)
 
+        # if k=1, make them 1D
+        if k == 1:
+            PA = PA.reshape(
+                -1,
+            )
```

Comment:
I should correct the format. 

Also:
I did not create a separate test function for `k=1`, and just make them 1D. However, I just realized that in `scrump`, this function gets 2D array when `k=1`. The same happens in `_stump`. So, I think I need to test `_merge_topk_PI` for k=1 for both 1D and 2D cases. Or, just do what @seanlaw suggested. create `_merge_topk_PI_1D` and `_merge_topk_PI_2D`. The former is only for `k=1`, and the latter is for `k >= 1` (note the equal sign)

Reply:
Why would you test the latter for `k=1` when it is 2D?

Reply:
Perhaps, you could use the same `squeeze` trick without checking the size of `PA`?

Reply:
> Perhaps, you could use the same squeeze trick without checking the size of PA?

Yeah! I think that would work. I have the same opinion. I think this should work.

Reply:
> Why would you test the latter for k=1 when it is 2D?

I think in `_stump` (which is a private function and the output is always 2D), we need to do `core._merge_topk_ρI` and its input is always 2D even when `k=1`.